literal/utils_test.go

```diff
@@ -738,3 +825,138 @@ func TestNewPrecisionTimestampTz(t *testing.T) {
+		{"2016-01-02T15:04:05", expr.NewPrimitiveLiteral(types.TimestampTz(1451747045000000), false), assert.NoError},
+		{"2016-01-02T15:04:05+00:00", expr.NewPrimitiveLiteral(types.TimestampTz(1451747045000000), false), assert.NoError},
+		{"2016-01-02T15:04:05+00:00", expr.NewPrimitiveLiteral(types.TimestampTz(1451747045000000), false), assert.NoError},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := NewTimestampTZFromString(tt.name)
+			if !tt.wantErr(t, err, fmt.Sprintf("NewTimestampFromString(%v)", tt.name)) {
+				return
+			}
+			assert.Equalf(t, tt.want, got, "NewTimestampFromString(%v)", tt.name)
+		})
+	}
+}
+
+func TestNewList(t *testing.T) {
+	i8Lit1 := NewInt8(1)
+	i8Lit2 := NewInt8(2)
+	i32Lit1, _ := NewInt32(1)
+	i32Lit2, _ := NewInt32(2)
+	listLiteral, _ := expr.NewLiteral[expr.ListLiteralValue]([]expr.Literal{i8Lit1, i8Lit2}, false)
+	int8Type := &types.Int8Type{Nullability: types.NullabilityRequired}
+	int32Type := &types.Int32Type{Nullability: types.NullabilityRequired}
+	int8ListType := &types.ListType{Type: int8Type, Nullability: types.NullabilityRequired}
+	int32ListType := &types.ListType{Type: int32Type, Nullability: types.NullabilityRequired}
+	listOfListType := &types.ListType{Type: int8ListType, Nullability: types.NullabilityRequired}
+	tests := []struct {
+		name       string
+		elements   []expr.Literal
+		expSuccess bool
+		litType    types.Type
+		wantErr    assert.ErrorAssertionFunc
+	}{
+		{"empty", []expr.Literal{}, false, nil, assert.Error},
+		{"i32List", []expr.Literal{i8Lit1, i32Lit2}, false, nil, assert.Error},
+		{"i8ListSingle", []expr.Literal{i8Lit1}, true, int8ListType, assert.NoError},
+		{"listOfListSingle", []expr.Literal{listLiteral}, true, listOfListType, assert.NoError},
+		{"i8List", []expr.Literal{i8Lit1, i8Lit2}, true, int8ListType, assert.NoError},
+		{"i32List", []expr.Literal{i32Lit1, i32Lit2}, true, int32ListType, assert.NoError},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := NewList(tt.elements)
+			if !tt.wantErr(t, err, fmt.Sprintf("NewList(%v)", tt.elements)) {
+				return
+			}
+			if tt.expSuccess {
+				want, err := expr.NewLiteral[expr.ListLiteralValue](tt.elements, false)
+				require.NoError(t, err)
+				assert.Equalf(t, want, got, "NewList(%v)", tt.elements)
+				assert.Equalf(t, tt.litType, got.GetType(), "NewList(%v)", tt.elements)
+			}
+		})
+	}
+}
+
+func TestNewDateFromString(t *testing.T) {
+	tests := []struct {
+		name    string
+		value   string
+		want    expr.Literal
+		wantErr assert.ErrorAssertionFunc
+	}{
+		{"empty", "", nil, assert.Error},
+		{"1970-01-01", "1970-01-01", expr.NewPrimitiveLiteral(types.Date(0), false), assert.NoError},
+		{"2021-01-01", "2021-01-01", expr.NewPrimitiveLiteral(types.Date(18628), false), assert.NoError},
+		{"2021-01-01T00:00:00Z", "2021-01-01T00:00:00Z", nil, assert.Error},
```

Comment:
We recently added support for strings like this elsewhere -- should we also support them here?

Reply:
can you please give reference?

Reply:
I missed this section is testing dates and not timestamps.  The test is fine as is.